Libraries/LibJS/Parser.cpp

```diff
@@ -1084,23 +1093,53 @@ NonnullRefPtr<Expression> Parser::parse_secondary_expression(NonnullRefPtr<Expre
             }
         }
         return create_ast_node<AssignmentExpression>(AssignmentOp::Assignment, move(lhs), parse_expression(min_precedence, associativity));
-    case TokenType::Period:
+    case TokenType::Period: {
         consume();
         if (!match_identifier_name())
             expected("IdentifierName");
-        return create_ast_node<MemberExpression>(move(lhs), create_ast_node<Identifier>(consume().value()));
+        auto rhs = create_ast_node<Identifier>(consume().value());
+        if (lhs->is_optional_member_expression())
+            return create_ast_node<OptionalMemberExpression>(move(lhs), move(rhs));
+        else
+            return create_ast_node<MemberExpression>(move(lhs), move(rhs));
```

Comment:
I know what you're going to say, no need for else here... but it looks a lot nicer this way, imo.

```cpp
if (lhs->is_optional_member_expression())
    return create_ast_node<OptionalMemberExpression>(move(lhs), move(rhs));
return create_ast_node<MemberExpression>(move(lhs), move(rhs));
```
:unamused:

Reply:
I agree that sometimes it does look nicer your way when choosing between two paths of equal semantic importance. :)